themes/tiddlywiki/vanilla/filter.tid

```diff
@@ -0,0 +1,34 @@
+title: $:/themes/tiddlywiki/vanilla/filter
+tags: $:/tags/Stylesheet
+
+\rules only filteredtranscludeinline transcludeinline macrodef macrocallinline macrocallblock conditional html
+
+<%if [{$:/state/story-river/filter}match[yes]] %>
+
+	.tc-story-river {
+		background: repeating-linear-gradient(
+			to right,
+			<<colour page-background>>,
+			<<colour page-background>> 10px,
+			<<colour tiddler-background>> 10px,
+			<<colour tiddler-background>> 20px
+		);
+	}
+
+	<%if [{$:/temp/search/input}!is[blank]] %>
+
+		<$list filter=<<tf.story-river.filter.match>> variable="filteredTiddler">
+
+			<$set name="CSSEscapedTitle" value={{{ [<filteredTiddler>escapecss[]] }}}>
+
+				[data-tiddler-title={{{ [<CSSEscapedTitle>] }}}].tc-tiddler-frame {
+					border: 3px solid <<colour primary>>;
+				}
+
+			</$set>
+
+		</$list>
+
+	<% endif %>
+
+<% endif %>

```

Comment:
The following code works fine. No CSS-escape, no extra new-lines.
The only differences are the **double quotes** that wrap the data-tiddler-title **value**.  `[data-tiddler-title="<<filteredTiddler>>"].tc-tiddler-frame {`

IMO `outline: 1px solid <<colour primary>>;` would be fine. [outline ](https://developer.mozilla.org/en-US/docs/Web/CSS/outline) has the advantage, that it does not change the box-metrics, which border does. 

```
<%if [{$:/state/story-river/filter}match[yes]] %>
	.tc-story-river {
		background: repeating-linear-gradient(
			to right,
			<<colour page-background>>,
			<<colour page-background>> 10px,
			<<colour tiddler-background>> 10px,
			<<colour tiddler-background>> 20px
		);
	}
	<%if [{$:/temp/search/input}!is[blank]] %>
		<$list filter=<<tf.story-river.filter.match>> variable="filteredTiddler">
				[data-tiddler-title="<<filteredTiddler>>"].tc-tiddler-frame {
					outline: 1px solid <<colour primary>>;
				}
		</$list>
	<% endif %>
<% endif %>
```